os/arch/arm/src/amebasmart/amebasmart_smp.c

```diff
@@ -201,15 +194,21 @@ void smp_init(void)
 #if ( CONFIG_SMP_NCPUS > 1 )
 	for (xCoreID = 0; xCoreID < CONFIG_SMP_NCPUS; xCoreID++) {
 		if (xCoreID == up_cpu_index()) {
-			// pmu_set_secondary_cpu_state(xCoreID, CPU1_RUNNING);
+			up_set_secondary_cpu_state(xCoreID, CPU_RUNNING);
 			continue;
 		}
-
 		err = psci_cpu_on(xCoreID, (unsigned long)__cpu1_start);
 		if (err < 0) {
-			printf("CPU%d: failed to boot: %d\n", (int)xCoreID, (int)err);
+			dbg("CPU%d: failed to boot: %d\n", (int)xCoreID, (int)err);
 		}
 	}
 #endif
-
+	/* We should check whether it is a warm boot here
+	   If yes, we should send SGI1
+	 */
+#ifdef CONFIG_CPU_HOTPLUG
+	if (up_get_secondary_cpu_state(1) == CPU_WAKE_FROM_SLEEP) {
```

Comment:
Can we have #define for core id instead of hardcoding?

Reply:
Done~